From ff04a00d75aac7b7f6f21816908bf1917846fb15 Mon Sep 17 00:00:00 2001
From: Leonardo Donelli <learts92@gmail.com>
Date: Fri, 10 May 2013 01:47:37 +0200
Subject: [PATCH] Set better window size

Make sure the window is not too big for the screen size.
The previous version did that in a way that didn't work
with some images size, particularly on 16:9 displays.
See https://bugzilla.gnome.org/show_bug.cgi?id=630512
---
 src/eog-window.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/eog-window.c b/src/eog-window.c
index ccf9564..f4f25ef 100644
--- a/src/eog-window.c
+++ b/src/eog-window.c
@@ -1244,13 +1244,11 @@ eog_window_obtain_desired_size (EogImage  *image,
 		if ((img_width + deco_width > screen_width) ||
 		    (img_height + deco_height > screen_height))
 		{
-			double factor;
+			double width_factor, height_factor, factor;
 
-			if (img_width > img_height) {
-				factor = (screen_width * 0.75 - deco_width) / (double) img_width;
-			} else {
-				factor = (screen_height * 0.75 - deco_height) / (double) img_height;
-			}
+			width_factor = (screen_width * 0.75 - deco_width) / (double) img_width;
+			height_factor = (screen_height * 0.75 - deco_height) / (double) img_height;
+			factor = MIN (width_factor, height_factor);
 
 			img_width = img_width * factor;
 			img_height = img_height * factor;
-- 
1.8.1.2

